src/module_3/module_3.md

```diff
@@ -0,0 +1,891 @@
+      <td>3</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>2</td>
+      <td>33</td>
+      <td>42.0</td>
+      <td>31.134054</td>
+      <td>30</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>3466586718340</td>
+      <td>2</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>2</td>
+      <td>33</td>
+      <td>42.0</td>
+      <td>31.134054</td>
+      <td>30</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>3481384026244</td>
+      <td>4</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>2</td>
+      <td>33</td>
+      <td>42.0</td>
+      <td>31.134054</td>
+      <td>30</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>3291363377284</td>
+      <td>2</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>2</td>
+      <td>33</td>
+      <td>42.0</td>
+      <td>31.134054</td>
+      <td>30</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>3479090790532</td>
+      <td>3</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>2</td>
+      <td>33</td>
+      <td>42.0</td>
+      <td>31.134054</td>
+      <td>30</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 24 columns</p>
+</div>
+
+
+
+### Training the model
+
+
+```python
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import SGDClassifier
+from sklearn.linear_model import RidgeClassifier
+```
+
+
+```python
+df['outcome'].value_counts()
+```
+
+
+
+
+    outcome
+    0    2132624
+    1      31329
+    Name: count, dtype: int64
+
+
+
+
+```python
+X = df.drop('outcome', axis=1)
+y = df['outcome']
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, test_size=0.4, random_state=9)
```

Comment:
stratify here isn't gonna make a big difference tbh, it's more used when you have a lot of categoriues, some of them very underrepresented

Reply:
More important (actually probably the most important thing of this exercise): I think this train / test split doesn't work for different reasons. First of all you are putting products of the same order in train and test at the same time. This is wrong since you are leaking data from train to test (if a concrete product was bought for an order, the rest of products have probably higher chances of being bought, but you can't know this in runtime). 
Also, you are ignoring the time component: you are putting contemporary orders at the same time in train and test. 
In general, the best way of designing a train test strategy is to think of how are you going to use the model, and what data is gonna be available for training and for testing. If you think of this you'll realise that in runtime we are only going to have available these orders which happened in the PAST and we are gonna predict on orders happening in the FUTURE. This is not what your train / test reflect since you are putting orders happening at the same time either in train and test.